### PR TITLE
Replaced manual escaping with Jackson

### PR DESCRIPTION
**Pull request by jozemlakar**
1. I have removed the manual escaping of new line
2. Escaping is now done by Jackson (only used for String)
3. ObjectMapper is now created when first used. According to this [article](http://stackoverflow.com/questions/3907929/should-i-declare-jacksons-objectmapper-as-a-static-field) it is thread-safe
4. Jackson adds double quotes to start and end of the stringified String object. These are removed using regex to fit the existing code.

### CODE REVIEW

**Comment by jozemlakar** [comment]
I have no idea why the build failed. Feeder timeout...?

File: gatling-core/src/main/scala/io/gatling/core/json/Json.scala [hunk 1]
```diff
@@ -45,8 +49,8 @@ object Json {
   }
 
   private def writeString(s: String, rootLevel: Boolean) = {
-    val escapedLineFeeds = s.replace("\n", "\\n")
-    if (rootLevel) fast"$escapedLineFeeds" else fast""""$escapedLineFeeds""""
+    val escapedLineFeeds = objectMapper.writeValueAsString(s).replaceAll("^\"|\"$", "")
```
Comment (slandelle) [inline gatling-core/src/main/scala/io/gatling/core/json/Json.scala]: Why do you need to replace some chars?

File: gatling-core/src/main/scala/io/gatling/core/json/Json.scala [hunk 2]
```diff
@@ -45,8 +49,8 @@ object Json {
...
+    if (rootLevel) fast"$escapedLineFeeds" else fast""""$escapedLineFeeds""""
```
Comment (slandelle) [inline gatling-core/src/main/scala/io/gatling/core/json/Json.scala]: trailing spaces

File: gatling-core/src/main/scala/io/gatling/core/json/Json.scala (hunk 1 above)
Comment (jozemlakar) [inline gatling-core/src/main/scala/io/gatling/core/json/Json.scala]: Jackson (correctly) adds double quotes to string as it is stringified.
`This is my string` will result in `"This is my string"` 

The purpose is of course that it aims to achieve the final goal being:
`key: "This is my string"` 

However in gatling double quotes are added later (ie writeMap) irrespective of the type. And so we would end up with `key: ""This is my string""` which is wrong. The reqex removes the double quoted at both ends.

File: gatling-core/src/main/scala/io/gatling/core/json/Json.scala (hunk 2 above)
Comment (jozemlakar) [inline gatling-core/src/main/scala/io/gatling/core/json/Json.scala]: Removed.

File: gatling-core/src/main/scala/io/gatling/core/json/Json.scala [hunk 3]
```diff
@@ -21,10 +21,14 @@ import scala.collection.JavaConverters._
...
 import com.dongxiguo.fastring.Fastring.Implicits._
 
+import com.fasterxml.jackson.databind.ObjectMapper
+
 object Json {
 
   def stringify(value: Any, isRootObject: Boolean = true): String =
     fastringify(value, isRootObject).toString()
+  
+  private val objectMapper = new ObjectMapper
```
Comment (slandelle) [inline gatling-core/src/main/scala/io/gatling/core/json/Json.scala]: It saddens me to have a mutable stateful singleton :(
We've worked hard in Gatling 2 to remove those that were making testing very difficult and brittle.

File: gatling-core/src/main/scala/io/gatling/core/json/Json.scala [hunk 4]
```diff
@@ -45,7 +49,7 @@ object Json {
   }
 
   private def writeString(s: String, rootLevel: Boolean) = {
-    val escapedLineFeeds = s.replace("\n", "\\n")
+    val escapedLineFeeds = objectMapper.writeValueAsString(s).replaceAll("^\"|\"$", "")
```
Comment (slandelle) [inline gatling-core/src/main/scala/io/gatling/core/json/Json.scala]: It's really sad we have to reprocess the Jackson output :(

File: gatling-core/src/main/scala/io/gatling/core/json/Json.scala (hunk 3 above)
Comment (jozemlakar) [inline gatling-core/src/main/scala/io/gatling/core/json/Json.scala]: I agree, The alternative would have been to create the mapper for the instance. I will try and fix this.

File: gatling-core/src/main/scala/io/gatling/core/json/Json.scala (hunk 4 above)
Comment (jozemlakar) [inline gatling-core/src/main/scala/io/gatling/core/json/Json.scala]: I absolutely agree. And I believe that Jackson got it right. We should fix the way gatling build JSON probably. I tried to keep the change contained. Please decide.

File: gatling-core/src/main/scala/io/gatling/core/json/Json.scala (hunk 3 above)
Comment (slandelle) [inline gatling-core/src/main/scala/io/gatling/core/json/Json.scala]: Don't do that, recreating the ObjectMapper on every call is a perf killer :(

File: gatling-core/src/main/scala/io/gatling/core/json/Json.scala (hunk 4 above)
Comment (slandelle) [inline gatling-core/src/main/scala/io/gatling/core/json/Json.scala]: > We should fix the way gatling build JSON probably.

We can't do that. It's intended to support re-injecting a captured JSON tree in a node position in a Gatling EL template.

Comment (jozemlakar) [inline gatling-core/src/main/scala/io/gatling/core/json/Json.scala]: I agree with all of above. The only way to get out of this mess is to copy the escaping mechanism from Jackson to gatling. Or to find one in Scala.

**Comment by slandelle** [comment]
@jozemlakar I'm very sorry with the very long delay :(

I'm going to cherry-picked this and build on top of it

Thanks!

**Pull request closed without merging**